### ISSUE TITLE
ERROR: rotation file has wrong creation timestamp

### ISSUE THREAD

**Issue by triwada** [postgrespro/pg_probackup#56]
There is a problem when using rotation of logs: backup stop with errror

```
pg_probackup-10 backup -B /vol-data1/backup --instance node1 -b FULL --stream -j 2 -C --delete-expired --delete-wal --log-level-console=INFO


INFO: Backup start, pg_probackup version: 2.0.26, backup ID: PQ****, backup mode: full, instance: node1, stream: true, remote: false
WARNING: This PostgreSQL instance was initialized without data block checksums. pg_probackup have no way to detect data block corruption without them. Reinitialize PGDATA with option '--data-checksums'.
ERROR: rotation file "/vol-data1/backup/log/node1_error.log.rotation" has wrong creation timestamp "
"
WARNING: Backup PQ**** is running, setting its status to ERROR

```

Problem file "/vol-data1/backup/log/node1_error.log.rotation" is empty.

pg_probackup.conf
```
# Backup instance information
pgdata = /var/lib/pgsql/10/data/
#Connection parameters:
PGDATABASE = postgres
PGUSER = backup
PGHOST = 127.0.0.1
#Replica parameters:
master-host = node1
master-port = 5432
master-user = backup
master-db = postgres
replica-timeout = 10min
#Logging parameters:
log-level-console = OFF
log-level-file = INFO
log-rotation-age = 3d
log-filename = node1.log
log-directory = /vol-data1/backup/log
error-log-filename = node1_error.log
#Retention parameters:
retention-window = 1
#Compression parameters:
compress-algorithm = zlib
compress-level = 1
```

**Comment by gsmol** [comment]
@za-arthur, I think we do not handle this sutiation. Safest approach in this case, I think, is to assume that creation_time = now().

**Comment by za-arthur** [comment]
Hello @triwada ,

It is strange that "/vol-data1/backup/log/node1_error.log.rotation" is empty.
Do you run several instances of `pg_probackup` concurrently?

**Comment by gsmol** [comment]
@za-arthur, it was not possibe in 2.0.26 to run several backups concurrently

**Comment by triwada** [comment]
> Do you run several instances of pg_probackup concurrently?
No, backup starts through task in crontab at exact time.

**Comment by gsmol** [comment]
I will write test, covering this case

**Comment by gsmol** [comment]
added e7c3031f7f7266c9357a19f075c525f30fc00f18

**Comment by gsmol** [comment]
`======================================================================
ERROR: test_truncated_rotation_file (tests.logging.LogTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/logging.py", line 116, in test_truncated_rotation_file
    '--log-filename=pg_probackup.log'])
  File "tests/helpers/ptrack_helpers.py", line 733, in backup_node
    return self.run_pb(cmd_list + options, asynchronous, gdb, old_binary)
  File "tests/helpers/ptrack_helpers.py", line 619, in run_pb
    raise ProbackupException(e.output.decode('utf-8'), self.cmd)
ProbackupException: 
 ERROR: u'ERROR: cannot read creation timestamp from rotation file "/home/gsmol/git/postgres/contrib/pg_probackup/tests/tmp_dirs/logging/test_truncated_rotation_file/backup/log/pg_probackup.log.rotation"\n'
 CMD: ['/home/gsmol/git/postgres/contrib/pg_probackup/pg_probackup backup -B /home/gsmol/git/postgres/contrib/pg_probackup/tests/tmp_dirs/logging/test_truncated_rotation_file/backup -p 16335 -d postgres --instance=node -b full --stream --log-level-file=VERBOSE --log-filename=pg_probackup.log']

----------------------------------------------------------------------
Ran 1 test in 12.920s

FAILED (errors=1)
`

**Comment by gsmol** [comment]
@triwada, was there previously a pg_probakup crash, kernel panic or poweroff?

**Comment by gsmol** [comment]
@za-arthur, proposed patch:
https://pastebin.postgrespro.ru/?8b3a102dcf4452a3#Z0s0sVF5uCJkvNPBzko/Kcz2sbxHGQT6JxA/aZe079w=

**Comment by triwada** [comment]
> @triwada, was there previously a pg_probakup crash, kernel panic or poweroff?

Sometimes backup was terminated using `kill PID of pg_probackup process`

**Comment by gsmol** [comment]
> > @triwada, was there previously a pg_probakup crash, kernel panic or poweroff?
> 
> Sometimes backup was terminated using `kill PID of pg_probackup process`

Is there was a reason to do that?

**Comment by triwada** [comment]
>  reason to do that?
when it is necessary to run backup later then it starts by crontab, but it was already started

PS: pg_probackup have not command to stop running backup accurately

**Comment by gsmol** [comment]
Oh, I see.
Well you can send SIGINT next time, it will do just the same thing from your perspective, but more gracefully to internal structure of backup catalog.
Reported problem also will be fixed.

**Comment by gsmol** [comment]
Do you think, we must provide some sort of command to stop running backups?

**Comment by triwada** [comment]
For me it is enough `kill -SIGINT PID_of_pg_probackup_process`.

If there are commands for start, merge, delete of backup, then it is logical to have a option for stop of running backup.

**Comment by gsmol** [comment]
It appears that we also do not handle SIGTERM correctly

**Comment by gsmol** [comment]
Rotation file issue fixed: fc8453b6c7f3

**Comment by gsmol** [comment]
@triwada, requested fix released with 2.1.1:
https://github.com/postgrespro/pg_probackup/releases/tag/2.1.1
2.1.1 packages are available in repository

Can we close this issue?
